### PR TITLE
Add showAdminBar docs

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR documents the new `getShowAdminBar()` method and closes PRO-6086.
## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Spin up the docs site
2. Navigate to `http://localhost:5173/guide/custom-ui.html#toggling-the-admin-bar-visibility` and `http://localhost:5173/reference/modules/admin-bar.html#getshowadminbar-req`
## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/guide/custom-ui.md [hunk 1]
```diff
@@ -250,6 +250,79 @@ Do not use core actions as your `action` property value - this would lead to unp
 * For backward compatibility, this method can also be called with the `moduleName` passed as the first argument and the object as the second, but this is discouraged.
 :::
 
+## Toggling the admin-bar visibility
+
+There are times when you want to allow people that don't have editing or content creation permissions to log into your project site. In that case, it may be desirable to not display the admin-bar. The `@apostrophecms/admin-bar` module `getShowAdminBar()` method can be extended to return `false` which will hide the admin-bar for that role when logged in.
```
Comment (boutell) [inline docs/guide/custom-ui.md:255]: This is a great example, so I think you should add a note about how useful this is right of the box: you can set the "visibility" of pages and pieces to "login required" in order to restrict access to users with at least a guest account, without showing them the Apostrophe admin bar.

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/guide/custom-ui.md [hunk 2]
```diff
@@ -250,6 +250,79 @@ Do not use core actions as your `action` property value - this would lead to unp
 * For backward compatibility, this method can also be called with the `moduleName` passed as the first argument and the object as the second, but this is discouraged.
 :::
 
+## Toggling the admin-bar visibility
```
Comment (boutell) [inline docs/guide/custom-ui.md:253]: "Toggling the visibility of the admin bar" might read better?

File: docs/guide/custom-ui.md [hunk 3]
```diff
@@ -250,6 +250,79 @@ Do not use core actions as your `action` property value - this would lead to unp
...
+
+There are times when you want to allow people that don't have editing or content creation permissions to log into your project site. For example, the visibility of a page or a piece document can be set to `login required`, so only those with an account can view it. In that case, it may be desirable to not display the admin-bar. The `@apostrophecms/admin-bar` module `getShowAdminBar()` method can be extended to return `false` which will hide the admin-bar for that role when logged in.
```
Comment (boutell) [inline docs/guide/custom-ui.md:255]: Do we already call it the "admin-bar" everywhere even when it's not a module name? If so then by all means don't let me change it up

**Review by boutell — APPROVED** [review]
This is great, minor question about language which is only a suggestion

File: docs/guide/custom-ui.md (hunk 3 above)
Comment (BoDonkey) [reply docs/guide/custom-ui.md:255]: Yup - checked that first thing.

File: docs/guide/custom-ui.md (hunk 2 above)
Comment (BoDonkey) [reply docs/guide/custom-ui.md:253]: Changed with a hyphen added.

**Pull request merged**